### ISSUE TITLE
Wrong deserialisation of floats

### ISSUE THREAD

**Issue by andreabedini** [souffle-lang/souffle#2093]
Hi there,

reading floats out of a SQLite database does not seem to be working. 

Steps to reproduce:

1) create a sqlite database called `data.sqlite3` as follows

```sql
CREATE TABLE test(x FLOAT);
INSERT INTO test VALUES (3.141524);
```

2) run this souffle program

```datalog
.decl test(x:float)
.input test(IO=sqlite, dbname="./data.sqlite3")
.output test(IO=stdout)
```

3) observe the output

```
---------------
test
===============
4.20389539e-45
===============
```

4.20389539e-45 is rapresented in binary as the integer value 3 (see [float.exposed](https://float.exposed/0x00000003)). 

Looking at https://github.com/souffle-lang/souffle/blob/946af0a5d6986f010bf5aeb34b7731bbbb889086/src/include/souffle/io/ReadStreamSQLite.h#L74-L82 it looks like we are reading it as an integer rather than a float

Adding an additional case for floats

 ```cpp
        switch (ty[0]) {
            case 's': tuple[column] = symbolTable.encode(element); break;
            case 'i':
            case 'u':
            case 'f': tuple[column] = RamFloatFromString(element); break;
            case 'r': tuple[column] = RamSignedFromString(element); break;
            default: fatal("invalid type attribute: `%c`", ty[0]);
        }
```

doesn't seem to solve the issue.

Debugging step-by-step the above code, it looks like `tuple[column]` does get assigned the right value 3.141524. Perhaps there's an issue somewhere else?

**Comment by andreabedini** [comment]
`RamSignedFromString("3.141524")` returns the correct value, but there might be some bit-hackery going on. As soon as I assign that value to `tuple[column]` (which is of type RamDomain) it gets sliced to 3. I am not understanding the design here, how is `ramBitCast` different from `static_cast`?

**Comment by andreabedini** [comment]
Basically I just re-discovered the issue around typecasting between int and floats. 
I think I fixed the issue, see #2094. Thanks for bearing with me :D
